expression/builtin_compare_vec.go

```diff
@@ -715,11 +715,53 @@ func (b *builtinNEStringSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column
 }
 
 func (b *builtinGreatestTimeSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinGreatestTimeSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	dst, err := b.bufAllocator.get(types.ETTimestamp, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(dst)
+
+	sc := b.ctx.GetSessionVars().StmtCtx
+	dst.ResizeTime(n, false)
```

Comment:
How about swapping the usage of `dst` and `result` so that we don't need to copy data from`dst` into `result` after calculating.

Reply:
I will try it.

Reply:
ditto,  `dst` is not the same type as `result`.  In my opinion, they can not swap each other.